android/src/main/java/com/reactnativesnapyrrnsdk/SnapyrRnSdkModule.java

```diff
@@ -1,43 +1,46 @@
 package com.reactnativesnapyrrnsdk;
 
-import androidx.annotation.NonNull;
-import androidx.annotation.Nullable;
-
 import android.app.Activity;
+import android.content.Context;
 import android.util.Log;
 
-import java.util.Objects;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
 
-import com.facebook.react.bridge.AssertionException;
 import com.facebook.react.bridge.LifecycleEventListener;
 import com.facebook.react.bridge.Promise;
 import com.facebook.react.bridge.ReactApplicationContext;
 import com.facebook.react.bridge.ReactContextBaseJavaModule;
 import com.facebook.react.bridge.ReactMethod;
 import com.facebook.react.bridge.ReadableMap;
-import com.facebook.react.bridge.Arguments;
-import com.facebook.react.bridge.WritableNativeMap;
+import com.facebook.react.bridge.WritableMap;
 import com.facebook.react.module.annotations.ReactModule;
-
-
-import com.facebook.react.modules.core.DeviceEventManagerModule;
+import com.snapyr.sdk.Properties;
 import com.snapyr.sdk.Snapyr;
 import com.snapyr.sdk.Traits;
-import com.snapyr.sdk.Properties;
 import com.snapyr.sdk.http.ConnectionFactory;
-import com.snapyr.sdk.inapp.InAppCallback;
 import com.snapyr.sdk.inapp.InAppConfig;
-import com.snapyr.sdk.inapp.InAppMessage;
 
 @ReactModule(name = SnapyrRnSdkModule.NAME)
 public class SnapyrRnSdkModule extends ReactContextBaseJavaModule implements LifecycleEventListener {
     public static final String NAME = "SnapyrRnSdk";
     private Boolean snapyrConfigured = false;
     private Boolean activityCallbacksReplayed = false;
+    private WritableMap initialIntentMap = null;
+
+    private Context getCurrentContext() {
```

Comment:
Updated to use the current activity if available; otherwise fall back to the application context. This was required because broadcast receivers can be fired when the app is not running, and may start up the React application context in the background, even though there is no activity running.